test/e2e/node/stress_probe.go

```diff
@@ -0,0 +1,196 @@
+		probeArgs := []string{"serve-hostname", "--http", "--port", ""}
+		pod := livenessPodSpec("http", f.Namespace.Name, livenessProbe, ports, probeArgs...)
+		runLivenessTest(ctx, f, pod)
+	})
+
+	/*
+		Release: v1.28
+		Testname: Pod liveness probe stress test, using tcp socket, no restart
+		Description: A Pod is created with MANY containers with liveness probe on tcp socket 8080. Liveness probe MUST not fail to check health and the restart count should remain 0 for all containers.
+	*/
+	ginkgo.It("should *not* be restarted with a tcp:8080 liveness probe stress test with MANY containers", func(ctx context.Context) {
+		livenessProbe := v1.Probe{
+			ProbeHandler:        tcpSocketHandler(containerPort),
+			InitialDelaySeconds: 15,
+			FailureThreshold:    1,
+		}
+		ports := []v1.ContainerPort{{ContainerPort: int32(containerPort)}}
+		probeArgs := []string{"serve-hostname", "--tcp", "--port", "", "--http=false"}
+		pod := livenessPodSpec("tcp", f.Namespace.Name, livenessProbe, ports, probeArgs...)
+		runLivenessTest(ctx, f, pod)
+	})
+
+	/*
+		Release: v1.28
+		Testname: Pod liveness probe stress test, using grpc call, no restart
+		Description: A Pod is created with MANY containers with liveness probe on grpc service. Liveness probe MUST not fail to check health and the restart count should remain 0 for all containers.
+	*/
+	ginkgo.It("should *not* be restarted with a GRPC liveness probe stress test with MANY containers", func(ctx context.Context) {
+		livenessProbe := v1.Probe{
+			ProbeHandler: v1.ProbeHandler{
+				GRPC: &v1.GRPCAction{
+					Port:    containerPort,
+					Service: nil,
+				},
+			},
+			InitialDelaySeconds: probeTestInitialDelaySeconds,
+			TimeoutSeconds:      5, // default 1s can be pretty aggressive in CI environments with low resources
+			FailureThreshold:    1,
+		}
+		ports := []v1.ContainerPort{{ContainerPort: int32(containerPort)}}
+		probeArgs := []string{"grpc-health-checking", "--port", "", "--http-port", ""}
+		pod := livenessPodSpec("grpc", f.Namespace.Name, livenessProbe, ports, probeArgs...)
+		runLivenessTest(ctx, f, pod)
+	})
+})
+
+func livenessPodSpec(probeType, namespace string, livenessProbe v1.Probe, ports []v1.ContainerPort, args ...string) *v1.Pod {
+	pod := e2epod.NewAgnhostPod(namespace, "liveness-stress-"+string(uuid.NewUUID()), nil, nil, ports, args...)
+	pod.ObjectMeta.Labels = map[string]string{"test": "liveness-stress"}
+	pod.Spec.Containers[0].LivenessProbe = &livenessProbe
+	pod.Spec.Containers[0].Name = "agnhost-container-0"
+	pod.Spec.Containers = append(pod.Spec.Containers, make([]v1.Container, numContainers-1)...)
+	for idx := 0; idx < numContainers; idx++ {
+		name := "agnhost-container-" + strconv.Itoa(idx+1)
+		instance := *pod.Spec.Containers[0].DeepCopy()
+		pod.Spec.Containers[idx] = newContainer(name, probeType, idx, instance)
+	}
+	return pod
+}
+
+func newContainer(name, probeType string, containerIdx int, instance v1.Container) v1.Container {
```

Comment:
this function is super confusing because it works based on the knowledge of the `probeArgs` and where exactly has to fill each parameter

Reply:
what you may suggest here? What parameters will be better to pass here. I'm passing probeType because it's hard to understand what type of probes is enabled, will apriciate if you have suggestions for this part as well.